tests/post/steps/test_service_configuration.py

```diff
@@ -0,0 +1,229 @@
+        k8s_client, name, namespace, patch
+    )
+
+    assert response
+
+    patched_csc = read_namespace_config_map(
+        k8s_client, name, namespace
+    )
+    config_obj = yaml_load(patched_csc, name, namespace)
+    patched_value = traversed_dict(config_obj, path)
+
+    assert int(value) == patched_value, (
+        "Expected value {} for key {} in ConfigMap {} found in namespace {}, "
+        "got {}".format(value, path, name, namespace, patched_value)
+    )
+
+
+@when(parsers.parse("we apply the '{state}' salt state"))
+def apply_service_config(host, version, request, k8s_client, state):
+    ssh_config = request.config.getoption('--ssh-config')
+
+    cmd = [
+        'salt-run', 'state.sls', '{}'.format(state),
+        'saltenv=metalk8s-{}'.format(version)
+    ]
+    output = utils.run_salt_command(host, cmd, ssh_config)
+
+    assert output.exit_status == 0, \
+        'Failed applying salt-state {}: \nout: {}\nerr:'.format(
+            state,
+            output.stdout,
+            output.stderr
+        )
+
+
+# }}}
+
+
+# Then {{{
+
+
+@then(parsers.parse(
+    "we have '{value}' at '{path}' for '{service}' Deployment in namespace "
+    "'{namespace}'"))
+def get_deployments(
+    host,
+    version,
+    k8s_appsv1_client,
+    k8s_client,
+    request,
+    value,
+    path,
+    service,
+    namespace
+
+):
+    def _wait_for_deployment():
+        try:
+            response = k8s_appsv1_client.read_namespaced_deployment(
+                name=service, namespace=namespace
+            ).to_dict()
+        except Exception as exc:
+            pytest.fail(
+                "Unable to read {} Deployment with error: {!s}".format(
+                    service, exc
+                )
+            )
+        assert int(value) == traversed_dict(response, path), (
+            "Expected value {} for service {}, got {}".format(
+                value, service, traversed_dict(response, path)
+            )
+        )
+    utils.retry(
+        _wait_for_deployment,
+        times=10,
+        wait=5,
+        name="wait for available Deployments"
+    )
+
+
+# }}}
+
+
+# Helpers {{{
+
+
+def read_namespace_config_map(k8s_client, name, namespace):
+    try:
+        response = k8s_client.read_namespaced_config_map(
+            name, namespace
+        )
+    except Exception as exc:
+        pytest.fail(
+            "Unable to read {} ConfigMap  in namespace {} with error: {!s}"
+            .format(name, namespace, exc)
+        )
+    return response
+
+
+def patch_namespace_config_map(k8s_client, name, namespace, patch):
+    try:
+        response = k8s_client.patch_namespaced_config_map(
+            name, namespace, patch
+        )
+    except Exception as exc:
+        pytest.fail(
+            "Unable to patch ConfigMap {} in namespace {} with error {!s}"
+            .format(name, namespace, exc)
+        )
+    return response
+
+
+def yaml_load(full_csc, name, namespace):
```

Comment:
Minor: Maybe just rename this function as it's not used to load a "yaml" but specific to CSC maybe "csc_load"